[Nicole Joseph] iP

nicolej2122 opened the pull request:
###  Ashy ChatBot
_Ashy_ is your friendly NUS cat that also helps with 
 - management of your tasks
 - creation of events 
 - tracking deadlines! 

Start using the _Ashy_ chatbot today! All you need to do is :
1.  Download it from [here](https://github.com/nicolej2122/ip/releases/download/A-Jar/ip.jar)
2. Run the command java `-jar ip.jar` in the same folder as the jar file
3. You can now let _Ashy_  manage your tasks! 😸

 Features: 
- [x] Management of tasks, events and deadlines
- [x] Tasks saved to hard-disk automatically 
- [x]  Search for tasks 

> Download the _Ashy_ ChatBot today for **FREE**!

Here's the main method of the bot!
```
public static void main(String[] args) {
        new Duke().run();
    }
```

Commit:
Add Gradle support

nicolej2122 committed:
Add Greet, Echo and Exit commands

nicolej2122 committed:
Add addition and listing

nicolej2122 committed:
Add ability to mark tasks as done

nicolej2122 committed:
Add ToDo, Event and Deadline tasks

nicolej2122 committed:
Added Text UI Testing

nicolej2122 committed:
Error handling

nicolej2122 committed:
Add deletion of tasks

nicolej2122 committed:
Use java enums

nicolej2122 committed:
Update Expected output file

nicolej2122 committed:
Add Date Time Formatting

nicolej2122 committed:
Add ability to save tasks to hard disk

nicolej2122 committed:
Resolve merge conflict

nicolej2122 committed:
Update .gitignore

nicolej2122 committed:
Usage of more OOP

nicolej2122 committed:
Package code

nicolej2122 committed:
Add JUnit tests

nicolej2122 committed:
Create a JAR file

nicolej2122 committed:
Add documentation

nicolej2122 committed:
Reformat code to follow Coding Standard

nicolej2122 committed:
Add ability to find tasks

nicolej2122 committed:
Resolve merge conflict

nicolej2122 committed:
Resolve merge conflict

nicolej2122 committed:
Merge branch 'add-gradle-support'

nicolej2122 committed:
Add checkstyle

GraceWang2322 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,191 @@
+
+    }
+
+    void done(int listNumber) {
```
maybe use a method name that is easier to understand or more specific??

nicolej2122 replied:
the list, done and find methods were named according to their corresponding command in the chatbot

GraceWang2322 commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,94 @@
+            }
+        }
+        Path path = Paths.get(directory, "data");
+        boolean exists = Files.exists(path);
```
maybe change boolean variable name to isExists according to the coding standards?

GraceWang2322 commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+     * @param listNumber index of task to be deleted
+     * @param commands tasks in chatbot
+     */
+    public void remove(int listNumber, ArrayList<Task> commands){
```
maybe have a space before the "{"

GraceWang2322 commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+    public void find(String keyword, ArrayList<Task> commands) {
+        try {
+            ArrayList<Task> matchingTasks = new ArrayList<>();
+            boolean matchFound = false;
```
maybe change boolean variable name to isMatchFound

nicolej2122 replied:
Noted!

GraceWang2322 reviewed (commented):
The coding standards are well-followed and the bot is really cutomised!! 👍 just a few nits to review for boolean naming.

nicolej2122 committed:
Add GUI

nicolej2122 committed:
Use of varargs

EssWhyy commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,19 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + time.format(DateTimeFormatter.ofPattern("MMM dd yyyy h:mma")) + ")";
```
This line is slightly long (above the soft limit of 110 characters), perhaps wrap it across 2 lines instead?

nicolej2122 replied:
Will do! Thanks!

EssWhyy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,189 @@
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
```
This is a large group of imports, perhaps consider separating imports of similar subclasses? (e.g. have a line break between duke.exceptions and duke.task and so on.)

nicolej2122 replied:
I tried that, but it seems to cause a checkstyle violation. I think imports from the same package are all grouped together

EssWhyy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,189 @@
+        try {
+            boolean taskToAdd = false;
+            int taskIndex;
+            switch (command) {
```
I like the use of ENUMS for your switch block!

EssWhyy reviewed (commented):
Generally well done, very few coding standard issues. Good use of OOP, javadocs and junit tests throughout the application.

nicolej2122 committed:
Set up Continuous Integration (CI)

To enables better transparency and farsightedness in the process of software development and delivery

Add gradle.yml file to .github/workflow directory

nicolej2122 committed:
Update gradle.yml

nicolej2122 committed:
Use assertions

nicolej2122 committed:
Refactor to improve code quality

nicolej2122 committed:
Update to resources

nicolej2122 committed:
Add ability to update task description in list

nicolej2122 committed:
Merge branch 'master' into branch-A-Assertions

nicolej2122 committed:
Merge pull request #1 from nicolej2122/branch-A-Assertions

Use of assertions

nicolej2122 committed:
Merge branch 'master' into branch-A-CodeQuality

nicolej2122 committed:
Merge pull request #2 from nicolej2122/branch-A-CodeQuality

Refactor to improve code quality

nicolej2122 committed:
Refactor code to improve compliance to coding standards

nicolej2122 committed:
Improve the GUI

nicolej2122 committed:
Add screenshot

nicolej2122 committed:
Update screenshot

nicolej2122 committed:
Set theme jekyll-theme-cayman

nicolej2122 committed:
Add a User Guide

nicolej2122 committed:
Merge remote-tracking branch 'origin/master'

nicolej2122 committed:
Add a User Guide

nicolej2122 committed:
Update README.md

nicolej2122 committed:
Added images to User Guide

nicolej2122 committed:
More updates to User Guide

nicolej2122 committed:
More updates to User Guide

nicolej2122 committed:
Update images in User Guide

nicolej2122 committed:
Add Usage and Acknowledgments

nicolej2122 committed:
Update structure of code

nicolej2122 committed:
Merge remote-tracking branch 'origin/master'
